File: app/src/main/java/com/wade/MathParser/exception/MathFunctionNotFoundException.java
package com.wade.MathParser.exception;

public class MathFunctionNotFoundException extends MathParserException {

    private final String function;

    public MathFunctionNotFoundException(String src, int index, String function) {
        super(src, index, function == null ? "couldn't find function" : function + "() not found");
        this.function = function;
    }

    public MathFunctionNotFoundException(String src, int index, String function, String message) {
        super(src, index, "couldn't find function: " + message);
        this.function = function;
    }

    public String getFunction() {
        return function;
    }
}

File: app/src/main/java/com/wade/MathParser/exception/MathFunctionNotFoundException.kt
package com.wade.MathParser.exception

class MathFunctionNotFoundException : MathParserException {
    val function: String?

    constructor(src: String?, index: Int, function: String?) : super(
        src,
        index,
        if (function == null) "couldn't find function" else "$function() not found"
    ) {
        this.function = function
    }

    constructor(src: String?, index: Int, function: String?, message: String) : super(
        src,
        index,
        "couldn't find function: $message"
    ) {
        this.function = function
    }
}